src/Compilers/CSharp/Portable/Symbols/SymbolWithAnnotations.cs

```diff
@@ -580,6 +580,17 @@ public TypeSymbolWithAnnotations SetUnknownNullabilityForReferenceTypesIfNecessa
                 this.SetUnknownNullabilityForReferenceTypes();
         }
 
+        public TypeSymbolWithAnnotations WithTopLevelNonNullability()
+        {
+            var typeSymbol = TypeSymbol;
+            if (IsNullable == false || !typeSymbol.IsReferenceType)
```

Comment:
>!typeSymbol.IsReferenceType [](start = 39, length = 27)

Should we do anything special for an unconstrained type parameter?

Reply:
Chuck fixed this in subsequent PR (https://github.com/dotnet/roslyn/pull/28282/files#diff-f6ce7c46b959e1ba9fd893d757ad7e2bR584)